This patch file is from Qualcomm Technologies, Inc.
and is provided under the following license:
Copyright (c) Qualcomm Technologies, Inc. and/or its subsidiaries.
SPDX-License-Identifier: BSD-3-Clause-Clear

From 29571a94bc90f70819a465f37ee9208915d92423 Mon Sep 17 00:00:00 2001
From: Sagar Kothawar <kothawar@qti.qualcomm.com>
Date: Tue, 21 Jul 2026 06:16:36 +0000
Subject: [PATCH] Subject: [PATCH] service_lan.sh: skip WAN interface in
bringup_ethernet_interfaces Skip IP assignment (169.254.1.x) on an
interface that is configured as WAN via syscfg wan_physical_ifname. This
prevents DHCP failure during LAN/WAN role switch on Echo platform

Signed-off-by: Sagar Kothawar <kothawar@qti.qualcomm.com>
---
 source/scripts/init/service.d/service_lan.sh | 7 ++++
 1 file changed, 7 insertions(+)

Index: git/source/scripts/init/service.d/service_lan.sh
===================================================================
--- git.orig/source/scripts/init/service.d/service_lan.sh
+++ git/source/scripts/init/service.d/service_lan.sh
@@ -158,6 +158,10 @@ bringup_ethernet_interfaces() {
    IFS=, read -r -a a <<< "$SYSCFG_lan_ethernet_physical_ifnames"
    for loop in "${a[@]}"
    do
+      # Skip if this interface is configured as WAN
+      wan_iface=$(syscfg get wan_physical_ifname)
+      [ "$loop" = "$wan_iface" ] && continue
+
       #Prevents clearing Ethernet interface IPs on the echo
       if [ "$BOX_TYPE" != "echo" ];then
          ifconfig "$loop" 0.0.0.0
@@ -696,6 +700,9 @@ service_init ()
    # use the physical ethernet names
    if [ -z "$SYSCFG_lan_ethernet_virtual_ifnums" ] ; then
       LAN_IFNAMES="$SYSCFG_lan_ethernet_physical_ifnames"
+      # Remove WAN interface from LAN_IFNAMES if wan_physical_ifname is set
+      wan_iface=$(syscfg get wan_physical_ifname)
+      [ -n "$wan_iface" ] && LAN_IFNAMES=$(echo "$LAN_IFNAMES" | tr ' ' '\n' | grep -v "^${wan_iface}$" | tr '\n' ' ' | xargs)
     else
        # in this architecture, vlans are used to separate the physical
        # ethernet switch into a wan port and lan ports.

-- 
2.34.1
